backends/s3/s3.go

```diff
@@ -273,52 +252,59 @@ func New(ctx context.Context, opt Options) (*Backend, error) {
 	ctx, cancel := context.WithTimeout(ctx, opt.InitTimeout)
 	defer cancel()
 
-	creds := credentials.NewStaticCredentialsProvider(opt.AccessKey, opt.SecretKey, "")
-	cfg, err := s3config.LoadDefaultConfig(
-		ctx,
-		s3config.WithCredentialsProvider(creds),
-		s3config.WithRegion(opt.Region),
-		s3config.WithHTTPClient(hc))
+	cfg := &minio.Options{
+		Creds: credentials.NewStaticV4(opt.AccessKey, opt.SecretKey, ""),
+		//Secure:    true,
```

Comment:
I guess this needs to depend on if the URL starts with http or https?

Reply:
Seems so indeed, I tested against an AWS s3 bucket with the following endpoint variations:
- s3.eu-central-1.amazonaws.com:443/
- s3.eu-central-1.amazonaws.com:80/
- s3.eu-central-1.amazonaws.com/

Worked fine over 'no port' + port 80, but seemed to stop functioning with port 443 specified. Likely due to a protocol mismatch.

Perhaps enforce or allow endpoints in opt.EndpointURL to have protocols specified, then set the minIO Client's 'Secure' option based on the protocol identified in the opt.EndpointURL?

Reply:
I updated the file to read the URL's scheme to determine the value of Secure. I then pass the URL with `https?:(//)?` stripped out to Minio.